src/pages/faq/comparison/pulumi.mdx

```diff
@@ -0,0 +1,129 @@
+
+# Side by Side
+
+To showcase the power of the abstraction provided by nitric here is a showcase of a nitric program with an equivalent Pulumi program with application code.
+
+<CodeGroup compare>
+
+```javascript {{title: 'Nitric', label: 'handle.ts' }}
+import * as nitric from '@nitric/sdk'
+
+const bucket = nitric.bucket('my-bucket').for('reading', 'writing')
+
+bucket.on('create', '*', async (ctx) => {
+  console.log(ctx.file.key, 'was created')
+})
+```
+
+```javascript {{title: 'Pulumi', label: 'lib/handle-stack.ts'}}
+import * as pulumi from '@pulumi/pulumi'
+import * as aws from '@pulumi/aws'
+import * as fs from 'fs'
+import * as mime from 'mime'
+import * as path from 'path'
+
+const config = new pulumi.Config()
+const region = config.require('aws:region')
+
+const bucket = new aws.s3.Bucket('s3-bucket')
+
+const role = new aws.iam.Role('lambdaRole', {
+  assumeRolePolicy: JSON.stringify({
+    Version: '2012-10-17',
+    Statement: [
+      {
+        Action: 'sts:AssumeRole',
+        Effect: 'Allow',
+        Principal: {
+          Service: 'lambda.amazonaws.com',
+        },
+      },
+    ],
+  }),
+})
+
+const policy = new aws.iam.RolePolicy('lambdaPolicy', {
+  role: role,
+  policy: JSON.stringify({
+    Version: '2012-10-17',
+    Statement: [
+      {
+        Action: [
+          'logs:CreateLogGroup',
+          'logs:CreateLogStream',
+          'logs:PutLogEvents',
+        ],
+        Effect: 'Allow',
+        Resource: 'arn:aws:logs:*:*:*',
+      },
+      {
+        Action: ['s3:PutObject'],
+        Effect: 'Allow',
+        Resource: `${bucket.arn}/*`,
+      },
+    ],
+  }),
+})
+
+const lambdaFunction = new aws.lambda.Function('s3UploaderLambda', {
+  runtime: 'nodejs14.x',
+  code: new pulumi.asset.AssetArchive({
+    '.': new pulumi.asset.FileArchive('./'),
+  }),
+  timeout: 10,
+  handler: 'index.handler',
+  role: role.arn,
+  environment: {
+    variables: {
+      BUCKET_NAME: bucket.bucket,
+    },
+  },
+})
+
+export const bucketName = bucket.bucket
+export const lambdaFunctionName = lambdaFunction.name
+```
+
+```javascript {{title: 'Pulumi', label: 'src/index.js'}}
+const AWS = require('aws-sdk')
+
+exports.handler = async (event) => {
+  const {
+    s3: { object },
+  } = event
+  console.log(object.key, 'was created')
+}
+```
+
+</CodeGroup>
+
+> In the pulumi example we get control over absolutely every facet of our infrastructure which is great, but for the majority of use cases a repeatable pattern of deployment is enough and scales much better than applying fine grained configuration every time we need to define a new deployment (with Pulumi this would typically be done by sharing infrastructure as libraries e.g. Pulumi Crosswalks).
```

Comment:
```suggestion
<Note>In the pulumi example we get control over absolutely every facet of our infrastructure which is great, but for the majority of use cases a repeatable pattern of deployment is enough and scales much better than applying fine grained configuration every time we need to define a new deployment (with Pulumi this would typically be done by sharing infrastructure as libraries e.g. Pulumi Crosswalks).</Note>
```

Reply:
done